core/src/main/scala/com/github/mrpowers/spark/fast/tests/ProductUtil.scala

```diff
@@ -1,35 +1,48 @@
 package com.github.mrpowers.spark.fast.tests
 
 import com.github.mrpowers.spark.fast.tests.ufansi.Color.{DarkGray, Green, Red}
+import com.github.mrpowers.spark.fast.tests.ufansi.FansiExtensions.StrOps
 import org.apache.commons.lang3.StringUtils
 import org.apache.spark.sql.Row
-import com.github.mrpowers.spark.fast.tests.ufansi.FansiExtensions.StrOps
-object DataframeUtil {
 
-  private[mrpowers] def showDataframeDiff(
+import scala.reflect.ClassTag
+
+object ProductUtil {
+  private[mrpowers] def productOrRowToSeq(product: Any): Seq[Any] = {
+    product match {
+      case null       => Seq.empty
+      case r: Row     => r.toSeq
+      case p: Product => p.productIterator.toSeq
+      case _          => throw new IllegalArgumentException("Only Row and Product types are supported")
+    }
+  }
+  private[mrpowers] def showProductDiff[T: ClassTag](
```

Comment:
We can now extend this to create a compare table for any T that has an API to  convert it to a Seq of value. like PrettyTable in python